home/home2.2/image.java

```diff
@@ -0,0 +1,184 @@
+                    return Constants.sizeProblem;
+                }       
+                for (int i = 0; i <SplitedLine.length; i++) {
+                    Pixel pix=new Pixel(SplitedLine[i]);
+                    if (!pix.isValid()) {
+                        return Constants.colorProblem;
+                    }
+                    pixels[i]=pix;
+                }
+                if (indexCounter == Constants.MatrixSize) {
+                    return Constants.sizeProblem;
+                }
+                lines[indexCounter]=pixels;
+                indexCounter++;
+            }    
+        } catch (Exception e) {}
+        return Constants.Fine;
+    }
+
+    public void changeMatrix(Pixel favouriteColor, Pixel hatedColor){
+        for (int i = 0; i < Constants.MatrixSize; i++) {
+            for (int j = 0; j < Constants.MatrixSize; j++) {
+                if (lines[i][j].compareColor(favouriteColor)) {
+                    if (i > 0) {
+                        lines[i - 1][j].setColor(favouriteColor);
+                    }
+                    if (j > 0) {
+                        lines[i][j - 1].setColor(favouriteColor);
+                    }
+                }
+                if (lines[i][j].compareColor(hatedColor) && hatedColor.isValid()) {
+                    lines[i][j].setColor(favouriteColor);    
+                }
+            }
+        }
+    }
+    
+    public void saveFile() {
+    try (BufferedWriter fileEdit = new BufferedWriter(new FileWriter("Output" + pixelMatrixFile))) {
+        for (int i = 0; i < Constants.MatrixSize; i++) {
+            String lineStr = arrayToString(lines[i]);
+            if (!lineStr.isEmpty()) {
+                fileEdit.write(lineStr);
+                if (i < Constants.MatrixSize - 1) {
+                    fileEdit.newLine();
+                }
+            }
+        }
+    } catch (Exception e) {}
+    }
+
+    private String arrayToString(Pixel[] line) {
+        StringBuilder sb = new StringBuilder();
+        for (Pixel s : line) {
+            if (sb.length() > 0) {
+                sb.append(" ");
+            }
+            sb.append(s.toString());
+        }
+        return sb.toString();
+    } 
+}
+
+public class image {
+    public static void main(String[] args) {
+        if (args.length < 2) {
+            System.out.println("Error:Please provide all arguments");
+            return;
+        }
+
+        String filename = args[0];
+        String color = args[1];
+        String HatedColor = (args.length > 2) ? args[2] : "";
+        Pixel colorPicked=new Pixel(color);
+        Pixel colorOfHate=new Pixel(HatedColor);
+        if(!colorPicked.isValid()){
+            System.out.println("Please, enter existing color for example 255,97,105");
+            return;
+        }
+        ImageMatrix image = new ImageMatrix(filename);
+        int loadResult=image.loadMatrix();
+        if(loadResult!=Constants.Fine){
```

Comment:
We can use a single switch to cover Fine case with
```
        image.changeMatrix(colorPicked, colorOfHate);
        image.saveFile();
        System.out.println("Image was changed successfully");
```
as a handler for that happy case